app/src/actioncommands.cpp

```diff
@@ -563,6 +568,186 @@ void ActionCommands::duplicateKey()
     mEditor->layers()->notifyAnimationLengthChanged();
 }
 
+
+void ActionCommands::manipulateRange()
+{
+    int loopBegin = 1, loopEnd = 2; // default values
+    if (mEditor->playback()->isRangedPlaybackOn())
+    {   // defined range values if available
+        loopBegin = mEditor->playback()->getRangedStartFrame();
+        loopEnd = mEditor->playback()->getRangedEndFrame();
+    }
+    LayerManager* layerMgr = mEditor->layers();
+    int lIndex = layerMgr->currentLayer()->type();
+    //   BITMAP = 1     VECTOR = 2  //not yet      SOUND = 4     CAMERA = 5
+    if (lIndex == 1 || lIndex == 2) //not yet || lIndex == 4 || lIndex == 5)
+    {
+        CopyMultiplekeyframesDialog* cd = new CopyMultiplekeyframesDialog(layerMgr, loopBegin, loopEnd, new QWidget);
+        cd->exec();
+
```

Comment:
Again, I don't understand. It may come down to me being self taught in C++, but to me the coding is progressing with a strict  (and sound) logic. 
But still - I'm willing to learn.

Reply:
In this case, insert `if (cd->result() != QDialog::Accepted) { return; }` here, then you can simplify `if (cd->result() == QDialog::Accepted && !cd->getValidity())` to `if (!cd->getValidity())` and the `if (cd->result() == QDialog::Accepted)` statement can be removed completely.